Cinnamon 5.6

bobby285271 opened the pull request:
**Update:**
Please also read https://www.linuxmint.com/rel_vera_cinnamon_whatsnew.php

---

This PR updates Cinnamon to 5.6, which is part of Linux Mint 21.1.

#### Lots of cosmetic changes!

Cosmetic changes are "design changes" (e.g. what theme, color, etc to use). Those changes are made by Linux Mint and we don't do random modifications, report to upstream if you don't like the changes. For items marked with *, you are only affected if you didn't set the related keys explicitly before and you can go back to old settings by using `gsettings`.

The following change set are generated from mint-artwork 1.6.5, cinnamon-common 5.6.2, nemo 5.6.0, mint-y-icons 1.6.3, things can still change in follow-up updates.

- The default cursor theme is now a mint-modified Bibata-Modern-Classic (see [this issue](https://github.com/ful1e5/Bibata_Cursor/issues?q=Linux+Mint%2C+sponsoring+and+pull+requests)), provided by mint-artwork. See change log from [mint-artwork](http://packages.linuxmint.com/pool/main/m/mint-artwork/) (download the `*.tar.xz` file and go to `debian/` folder) for more info. Note that we are removing `vanilla-dmz` from the module.
- \* The default GTK theme and icon theme is now Mint-Y-Aqua, it is changed by mint-artwork.
- \* Home/trash/network desktop icons are no longer presented on desktop, this is announced in [Sep news](https://blog.linuxmint.com/?p=4408), still changed by mint-artwork.
- Nemo no longer highlight selected icons, this is announced in [Oct news](https://blog.linuxmint.com/?p=4424), changed by https://github.com/linuxmint/nemo/commit/fdd47db7a7399ae12e21912fa33bd81cd1ba0eac.
- \* There is a new cornerbar applet that replaces show-desktop, changed by https://github.com/linuxmint/cinnamon/commit/cd47f9aef5c383a108c5f77098d3c602b2e0bc86.
- The places icon set has been switched to a Papirus based one, now you might see yellow folder icons everywhere, changed by https://github.com/linuxmint/mint-y-icons/commit/7f804620de2955d7d91e60ddfa1bdfb1fcd967b0.
- The Mint-Y-Yellow theme is gone, changed by https://github.com/linuxmint/mint-y-icons/commit/41fcdb33cd845861e14ab595d0c27259b50459b9.

If some of your settings are "changed" / "broken" after this update, reconfigure them in the cinnamon-settings app.

bobby285271 committed:
cinnamon.cinnamon-menus: 5.4.0 -> 5.6.0

No changes.

https://github.com/linuxmint/cinnamon-menus/compare/5.4.0...5.6.0

bobby285271 committed:
cinnamon.xapp: 2.2.15 -> 2.4.1

https://github.com/linuxmint/xapp/compare/2.2.15...2.4.1

bobby285271 committed:
python3.pkgs.xapp: 2.2.2 -> 2.4.0

Switched to meson.

https://github.com/linuxmint/python3-xapp/compare/2.2.2...2.4.0

bobby285271 committed:
cinnamon.mint-artwork: 1.6.0 -> 1.6.8

Hided home/trash/network desktop icons by default.

Also changed their default logo settings (cs_info and menu), so what we
should modify to cinnamon-symbolic is now linuxmint-logo-ring-symbolic.
We are lucky, linuxmint-logo-ring-symbolic only appears in a few other
places and non of them are useful, menu@cinnamon.org/settings-override.json
simply won't work because cinnamon-common doesn't know that file.

Also set default cursor theme, this package now directly provides those
cursor themes. And, we now default to the Mint-Y-Aqua theme.

mkg20001 commented:
@bobby285271 for disabling @r-ryantm making prs we'd need to contact @ryantm (hi, could you please disable cinnamon.* for now?)

ryantm commented:
@mkg20001 Should be skipped now until further notice!

bobby285271 committed:
cinnamon.cinnamon-control-center: 5.4.7 -> 5.6.0

I removed the treewide /usr/share/locale sed, there are still
a few leftovers but non of them are useful (for test-endianess.c
we already fixed /usr/share/i18n/locales/, and we don't use gdm
so gdm-languages.c is not useful).

Also no longer depend on cinnamon-settings-daemon.

https://github.com/linuxmint/cinnamon-control-center/compare/5.4.7...5.6.0

bobby285271 committed:
cinnamon.cinnamon-translations: 5.4.2 -> 5.6.0

No changes.

https://github.com/linuxmint/cinnamon-translations/compare/5.4.2...5.6.0

bobby285271 committed:
cinnamon.cinnamon-session: 5.4.0 -> 5.6.0

https://github.com/linuxmint/cinnamon-session/compare/5.4.0...5.6.0

bobby285271 committed:
cinnamon.cinnamon-settings-daemon: 5.4.5 -> 5.6.0

No changes.

https://github.com/linuxmint/cinnamon-settings-daemon/compare/5.4.5...5.6.0

bobby285271 committed:
cinnamon.cinnamon-common: 5.4.12 -> 5.6.2

PR 11244 removed some of the hardcode paths:

- We now look for upload-system-info from PATH, but since we don't
"install" cinnamon.xapp, we still fix the path for it.
- We now look for cinnamon-control-center, gnome-control-center from
PATH. For the former, it is installed in the module and it cannot be
removed, so no need to fix the path. For the latter, it is not useful.
- cinnamon-launcher now looks for various fallback panels from PATH,
since they are optional so no need of fixing path.
- All *.desktop and *.service files are generated by Meson with correct
path now.

https://github.com/linuxmint/cinnamon/compare/5.4.12...5.6.2

bobby285271 committed:
cinnamon.cinnamon-screensaver: 5.4.4 -> 5.6.1

This seems to fix "Fractional scaling resets on resume after suspend".

https://github.com/linuxmint/cinnamon-screensaver/compare/5.4.4...5.6.1

bobby285271 committed:
cinnamon.xreader: 3.4.5 -> 3.6.0

No changes.

https://github.com/linuxmint/xreader/compare/3.4.5...3.6.0

bobby285271 committed:
cinnamon.cinnamon-control-center: Clean up

There is no test since 5.4.6, 23ba143 "wacom: Bring in line with
changes in muffin to fix support" removed the wacom-calibrator test.

All gio-unix-2.0 dep seemed to be properly declared. Still builds.

bobby285271 committed:
cinnamon.muffin: 5.4.7 -> 5.6.0

https://github.com/linuxmint/muffin/compare/5.4.7...5.6.0

bobby285271 committed:
cinnamon.cinnamon-desktop: 5.4.2 -> 5.6.0

https://github.com/linuxmint/cinnamon-desktop/compare/5.4.2...5.6.0

bobby285271 committed:
cinnamon.cjs: 5.4.1 -> 5.6.1

No changes.

https://github.com/linuxmint/cjs/compare/5.4.1...5.6.1

mkg20001 commented on pkgs/desktops/cinnamon/cinnamon-common/default.nix:
```diff
@@ -149,26 +151,16 @@ stdenv.mkDerivation rec {
 
     sed "s|/usr/share/sounds|/run/current-system/sw/share/sounds|g" -i ./files/usr/share/cinnamon/cinnamon-settings/bin/SettingsWidgets.py
 
-    sed "s|/usr/bin/upload-system-info|${xapp}/bin/upload-system-info|g" -i ./files/usr/share/cinnamon/cinnamon-settings/modules/cs_info.py
```
why are those getting removed?

bobby285271 replied:
https://github.com/linuxmint/cinnamon/pull/11244

bobby285271 commented:
Okay the cinnamon tests fails because http://packages.linuxmint.com/pool/main/m/mint-artwork/mint-artwork_1.6.2.tar.xz is 404 and 1.6.3 is out, it looks like upstream only keeps some specific versions :thinking:

bobby285271 committed:
xed-editor: 3.2.7 -> 3.2.8

https://github.com/linuxmint/xed/compare/3.2.7...3.2.8

bobby285271 committed:
cinnamon.nemo: 5.4.3 -> 5.6.0

https://github.com/linuxmint/nemo/compare/5.4.3...5.6.0

bobby285271 committed:
hypnotix: Fix launching with mpv 0.35.0

See hypnotix issue 254 for more info.

bobby285271 committed:
hypnotix: 2.9 -> 3.0

https://github.com/linuxmint/hypnotix/compare/2.9...3.0

bobby285271 committed:
rl-23.05: Mention cinnamon 5.6 update

bobby285271 committed:
cinnamon.pix: 2.8.7 -> 2.8.8

https://github.com/linuxmint/pix/compare/2.8.7...2.8.8

bobby285271 committed:
cinnamon.xviewer: 3.2.10 -> 3.2.11

https://github.com/linuxmint/xviewer/compare/3.2.10...3.2.11

bobby285271 committed:
cinnamon.mint-x-icons: Switch to tags, use stdenvNoCC

You don't need cc to build and install this.

bobby285271 committed:
cinnamon.mint-y-icons: Switch to tags, use stdenvNoCC

You don't need cc to build and install this.

bobby285271 committed:
cinnamon.mint-themes: Switch to tags, use stdenvNoCC

You don't need cc to build and install this.

bobby285271 committed:
timeshift: 22.06.5 -> 22.11.1

https://github.com/linuxmint/timeshift/compare/22.06.5...22.11.1

bobby285271 committed:
cinnamon.mint-y-icons: 1.6.1 -> 1.6.5

https://github.com/linuxmint/mint-y-icons/compare/1.6.1...1.6.5

bobby285271 commented:
So... here is how the new icons look like :sweat_drops: 

<img src="https://user-images.githubusercontent.com/20080233/203460518-0174e2b2-8bcb-4811-8388-31980f0c878e.png" width=50%>

<img src="https://user-images.githubusercontent.com/20080233/203460892-aadd627a-2eda-4cc5-bcbb-028c80fb05b6.png" width=50%>

bobby285271 committed:
cinnamon.mint-themes: 2.0.5 -> 2.0.7

https://github.com/linuxmint/mint-themes/compare/2.0.5...2.0.7

bobby285271 commented:
Per [my version checker](https://bobby285271.github.io/what-changed/003-cinnamon.html) all packages with major changes should have at least one new tag now. I plan to merge this bump in this weekend, if you wish to review this PR please do, thanks.

dotlambda commented:
> rl-23.05: Mention cinnamon 5.6 update

Why don't we include this in 22.11?
cc @NixOS/nixos-release-managers

bobby285271 commented:
:bow: Sorry I didn't consider this as release blocker since this is too recent, and I didn't get this in before branch off. Consider no mutter rebase and gjs rebase are evolved here, do you have specific bug fixes that you want to backport?

dotlambda commented:
> Since no mutter rebase and gjs rebase are involved here, do you have specific bug fixes that you want to backport?

No, but we could back port the entire PR. I don't use Cinnamon and have no idea how stable 5.6 is, so it's your call.

jonringer commented:
> No, but we could back port the entire PR. I don't use Cinnamon and have no idea how stable 5.6 is, so it's your call.

The main consideration is if it would "break" any behavior from the previous release. Since desktop platforms usually have a very large scope (high likelihood of regressions), I'm apprehensive about backporting.

bobby285271 commented:
@ofborg test cinnamon
@ofborg build hypnotix timeshift cinnamon.mint-artwork

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/breaking-changes-announcement-for-unstable/17574/17

mkg20001 commented:
@ryantm you can enable  cinnamon.* now

bobby285271 commented:
TBH r-ryantm won't have much chance to open PRs before I retired, I have some scripts that check updates twice a day and I batch everything in a PR and rebuild my laptop to test that at least once a day. You may check how this is done in [`pkgs/desktops/pantheon`](https://github.com/NixOS/nixpkgs/commits/master/pkgs/desktops/pantheon) in the past year :upside_down_face: 

But anyway I guess I want to mention here that I skipped 2 updates here:

- cinnamon 5.6.3: The only change is later reverted.
- warpinator 1.4.0: I guess it has an undefined 'self' in [auth.py](https://github.com/linuxmint/warpinator/blob/9f17136d4490b0089bb82a7f71f51acd0e9fff8b/src/auth.py#L111) that will prevent this from working.

ryantm commented:
@mkg20001 Re-enabled thanks.
